src/ui/components/NetworkMonitor/index.tsx

```diff
@@ -93,6 +101,16 @@ export const NetworkMonitor = ({
         let selectedRequest;
         if (selectedRequestId) {
           selectedRequest = data.find(request => request.id === selectedRequestId);
+          if (selectedRequest && loadedRegions) {
+            if (!isTimeInRegions(selectedRequest.point.time, loadedRegions.loaded)) {
+              // Unloaded requests should not be selectable.
+              //
+              // TODO I'm not sure this logic is appropriate for the view,
+              // especially a lower level view like NetworkMonitor, to decide.
+              // This risks "tearing" (where different views show different behavior).
+              selectedRequest = undefined;
```

Comment:
This is not great. We're still _selecting_ the request in our state; we're just overriding it in the view layer. This kind of follows the existing pattern (in the `useEffect` above) which I think is also something we should clean up, but maybe as a separate follow up PR?

Reply:
I think two places are currently doing the "selecting" here:
* The comment renders the secondary label using the `NetworkRequestPreview` which has a click-handler that selects the network monitor and shows the request details.
* The Network panel's internal `RequestTable` component has a row-click-handler that does the same.

For both of the above cases, I think...
1. Jumping to the request's timestamp is a good idea, but–
2. We shouldn't update the Redux state to select the request if it's not loaded.

Reply:
I'm going to go ahead and pull in this scope as well. I'm not _100% confident_ about the approach though so I'm going to tag @markerikson on it specifically for the Redux stuff 🙇🏼